src/proj_functions.jl

```diff
@@ -0,0 +1,211 @@
+    _inv!(xy, proj.rep)
+    if !radians
+        for i in eachindex(xy)
+            xy[i] = rad2deg(xy[i])
+        end
+    end
+    xy
+end
+
+@doc "Returns the inverse projection from XY to LatLon in the given projection (only supports 2 dimensions)" ->
+xy2lonlat(xy::Vector{Float64}, proj::Projection, radians::Bool=false) = xy2lonlat!(copy(xy), proj, radians)
+xy2lonlat(xy::Array{Float64,2}, proj::Projection, radians::Bool=false) = xy2lonlat!(copy(xy), proj, radians)
+
+@doc """
+Transform between geographic or projected coordinate systems
+
+Args:
+
+    src      - Source coordinate system definition
+    dest     - Destination coordinate system definition
+    position - An Nx2 or Nx3 array of coordinates to be transformed in place.
+               For geographic coordinate systems, the first two columns are
+               the *longitude* and *latitude*, in that order.
+    radians  - If true, treat geographic lon,lat coordinates as radians on
+               input and output.
+
+Returns:
+
+    position - Transformed position
+""" ->
+function transform2!(src::Projection, dest::Projection, position::Array{Float64,2}, radians::Bool=false)
+    !radians && is_latlong(src) && (position[:,1:2] = deg2rad(position[:,1:2]))
+    _transform2!(src.rep, dest.rep, position[:,1:2])
+    !radians && is_latlong(dest) && (position[:,1:2] = rad2deg(position[:,1:2]))
+    position
+end
+transform2(src::Projection, dest::Projection, position::Array{Float64,2}, radians::Bool=false) =
+    transform2!(src, dest, copy(position[:,1:2]), radians)
+
+function transform2!(src::Projection, dest::Projection, position::Vector{Float64}, radians::Bool=false)
+    !radians && is_latlong(src) && (position[1:2] = deg2rad(position))
+    _transform2!(src.rep, dest.rep, position)
+    !radians && is_latlong(dest) && (position[1:2] = rad2deg(position))
+    position
+end
+transform2(src::Projection, dest::Projection, position::Vector{Float64}, radians::Bool=false) =
+    transform2!(src, dest, copy(position), radians)
+
+function transform3!(src::Projection, dest::Projection, position::Array{Float64,2}, radians::Bool=false)
+    !radians && is_latlong(src) && (position[:,1:3] = deg2rad(position[:,1:3]))
+    _transform3!(src.rep, dest.rep, position[:,1:3])
+    !radians && is_latlong(dest) && (position[:,1:3] = rad2deg(position[:,1:3]))
+    position
+end
+transform3(src::Projection, dest::Projection, position::Array{Float64,2}, radians::Bool=false) =
+    transform3!(src, dest, copy(position[:,1:3]), radians)
+
+function transform3!(src::Projection, dest::Projection, position::Vector{Float64}, radians::Bool=false)
+    !radians && is_latlong(src) && (position[1:3] = deg2rad(position))
+    _transform3!(src.rep, dest.rep, position)
+    !radians && is_latlong(dest) && (position[1:3] = rad2deg(position))
+    position
+end
+transform3(src::Projection, dest::Projection, position::Vector{Float64}, radians::Bool=false) =
+    transform3!(src, dest, copy(position), radians)
```

Comment:
Okay, removed the multiplicity of transforms. Thanks for being persistent on this one. It took me a awhile to track down all the inefficiencies, which led to a much cleaner codebase in the end. See https://github.com/FugroRoames/Proj4.jl/pull/4#issuecomment-137867408 for a small benchmark